lightning/src/ln/channelmanager.rs

```diff
@@ -1775,6 +1775,103 @@ impl<ChanSigner: ChannelKeys, M: Deref> ChannelManager<ChanSigner, M> where M::T
 		PublicKey::from_secret_key(&self.secp_ctx, &self.our_network_key)
 	}
 
+	/// Restores a single, given channel to normal operation after a
+	/// ChannelMonitorUpdateErr::TemporaryFailure was returned from a channel monitor update
+	/// operation.
+	///
+	/// All ChannelMonitor updates up to and including highest_applied_update_id must have been
+	/// fully committed in every copy of the given channels' ChannelMonitors.
+	///
+	/// Note that there is no effect to calling with a highest_applied_update_id other than the
+	/// current latest ChannelMonitorUpdate and one call to this function after multiple
+	/// ChannelMonitorUpdateErr::TemporaryFailures is fine. The highest_applied_update_id field
+	/// exists largely only to prevent races between this and concurrent update_monitor calls.
```

Comment:
I think we're on the same page - you are passed a stream of monitor updates, you save them all to disk, you pass them to your remote watchtowers in order, and when the watchtower has accepted updates you call channel_monitor_updated with the applied updates. Once its up to date the Channel will activate again as it will see the update id is the same as the latest one. I added a comment describing that flow, let me know if its sufficient.

Reply:
Yes that's better, I will still insert between 2) and 3) "You keep calling update_monitor until getting a success or a PermanentFailure"

Reply:
"You" (being the user in this case) does not call update_monitor, "we" (being rust-lightning) do. I'm not sure what you're getting at here.

Reply:
Here, user should keep calling update_monitor between 2) and 3)?